packit_service/models.py

```diff
@@ -0,0 +1,197 @@
+    global session_instance
+    if session_instance is None:
+        url = (
+            f"postgres+psycopg2://{os.getenv('POSTGRESQL_USER')}"
+            f":{os.getenv('POSTGRESQL_PASSWORD')}@{os.getenv('POSTGRES_SERVICE_HOST', 'postgres')}"
+            f":5432/{os.getenv('POSTGRESQL_DATABASE')}"
+        )
+        engine = create_engine(url)
+        Session = sessionmaker(bind=engine)
+        session_instance = Session()
+    return session_instance
+
+
+# https://github.com/python/mypy/issues/2477#issuecomment-313984522 ^_^
+if TYPE_CHECKING:
+    Base = object
+else:
+    Base = declarative_base()
+
+
+class GitHubProject(Base):
+    __tablename__ = "github_projects"
+    id = Column(Integer, primary_key=True)
+    namespace = Column(String, index=True)
+    repo_name = Column(String, index=True)
+    pull_requests = relationship("PullRequest", back_populates="project")
+
+    @classmethod
+    def get_or_create(cls, namespace: str, repo_name: str) -> "GitHubProject":
+        session = get_sa_session()
+        project = (
+            session.query(GitHubProject)
+            .filter_by(namespace=namespace, repo_name=repo_name)
+            .first()
+        )
+        if not project:
+            project = cls()
+            project.repo_name = repo_name
+            project.namespace = namespace
+            session.add(project)
+            session.commit()
+        return project
+
+
+class PullRequest(Base):
+    __tablename__ = "pull_requests"
+    id = Column(Integer, primary_key=True)  # our database PK
+    pr_id = Column(
+        Integer, index=True
+    )  # GitHub PR ID - let's not make this PK since we can't control it
+    project_id = Column(Integer, ForeignKey("github_projects.id"))
+    project = relationship("GitHubProject", back_populates="pull_requests")
+    copr_builds = relationship("CoprBuild", back_populates="pr")
+
+    @classmethod
+    def get_or_create(cls, pr_id: int, namespace: str, repo_name: str) -> "PullRequest":
+        session = get_sa_session()
+        project = GitHubProject.get_or_create(namespace=namespace, repo_name=repo_name)
+        pr = session.query(PullRequest).filter_by(pr_id=pr_id).first()
+        if not pr:
+            pr = PullRequest()
+            pr.pr_id = pr_id
+            pr.project_id = project.id
+            session.add(pr)
+            session.commit()
+        return pr
+
+
+class CoprBuild(Base):
+    """ we create an entry for every target """
+
+    __tablename__ = "copr_builds"
+    id = Column(Integer, primary_key=True)
+    build_id = Column(String, index=True)  # copr build id
+    pr_id = Column(Integer, ForeignKey("pull_requests.id"))
+    pr = relationship("PullRequest", back_populates="copr_builds")
+    logs = Column(Text)
+    commit_sha = Column(String)
+    status = Column(String)
+    target = Column(String)
+    web_url = Column(String)  # URL to web ui
+    build_logs_url = Column(String)  # url to copr build logs
+    logs_url = Column(String)
```

Comment:
Might be good to have a more explicit documentation for these fields, for example to specify which service these URLs are pointing to: copr or Packit Service.